aries_vcx/src/protocols/issuance/issuer/state_machine.rs

```diff
@@ -236,39 +271,42 @@ impl IssuerSM {
         for (uid, message) in messages {
             match self.state {
                 IssuerFullState::Initial(_) => {
-                    if let A2AMessage::CredentialProposal(credential_proposal) = message {
-                        return Some((uid, A2AMessage::CredentialProposal(credential_proposal)));
+                    if let AriesMessage::CredentialIssuance(CredentialIssuance::ProposeCredential(_)) = &message {
+                        return Some((uid, message));
                     }
                 }
-                IssuerFullState::OfferSent(_) => match message {
-                    A2AMessage::CredentialRequest(credential) => {
-                        if credential.from_thread(&self.thread_id) {
-                            return Some((uid, A2AMessage::CredentialRequest(credential)));
+                IssuerFullState::OfferSent(_) => match &message {
+                    AriesMessage::CredentialIssuance(CredentialIssuance::RequestCredential(msg)) => {
+                        if matches_opt_thread_id!(msg, self.thread_id.as_str()) {
```

Comment:
This is slightly downgrade, as at least `thread` decorator is known to be always required to present on a message. The protocol can not start from `credential-request` message and is always reply to `credential-offer`. So I think it can be safely said it should always have a thread.

What do you think? Just to clarify, I am not suggesting to address it here.

Reply:
It was optional before, so I left it like that.
Per the RFC, it is accurate though. 
https://github.com/hyperledger/aries-rfcs/blob/main/features/0036-issue-credential/README.md#messages